platform/linux-generic/include/odp_atomic_internal.h

```diff
@@ -38,7 +38,7 @@ typedef struct ODP_ALIGNED(sizeof(void *)) {
  * @Note this is not the same as a plain boolean type.
  * _odp_atomic_flag_t is guaranteed to be able to operate on atomically.
  */
-typedef char _odp_atomic_flag_t;
+typedef int _odp_atomic_flag_t;
```

Comment:
According to `__atomic_clear()` definition, `_odp_atomic_flag_t` declaration should be changed to `bool`, not to int or char.